uniffi_bindgen/src/bindings/swift/templates/RustBufferHelper.swift

```diff
@@ -225,6 +225,75 @@ extension Bool: ViaFfi {
     }
 }
 
+extension Date: ViaFfiUsingByteBuffer, ViaFfi {
+    fileprivate static func read(from buf: Reader) throws -> Self {
+        let seconds: Int64 = try buf.readInt()
+        let microseconds: UInt32 = try buf.readInt()
+        if seconds >= 0 {
+            let delta = Double(seconds) + (Double(microseconds) / 1.0e9)
+            return Date.init(timeIntervalSince1970: delta)
+        } else {
+            let delta = Double(-seconds) + (Double(microseconds) / 1.0e9)
+            return Date.init(timeIntervalSince1970: -delta)
```

Comment:
I don't understand the double negation here (once in `Double(-seconds)` and then again on the result in `-delta`. Is it necessary, or does it help with precision or something like that?

Reply:
I was trying to match how the math works in the other languages, but it just makes for a confusing read in this case. Will update to simplify.